examples/v1alpha2/basic-tcp.yaml

```diff
@@ -0,0 +1,45 @@
+kind: Gateway
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: my-tcp-gateway
+spec:
+  gatewayClassName: my-tcp-gateway-class
+  listeners:
+  - name: foo
+    protocol: TCP
+    port: 8080
+    routes:
+      kinds:
+      - kind: TCPRoute
+  - name: bar
+    protocol: TCP
+    port: 8090
+    routes:
+      kinds:
+      - kind: TCPRoute
+---
+kind: TCPRoute
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: tcp-app-1
+spec:
+  parentRefs:
+  - name: my-tcp-gateway
+    sectionName: foo
```

Comment:
This section name feels off. Or maybe it is a new thing so it _feels_ that way. Just a comment and no action required.